fastapi/dependencies/utils.py

```diff
@@ -245,15 +245,31 @@ def is_scalar_sequence_field(field: ModelField) -> bool:
 def get_typed_signature(call: Callable[..., Any]) -> inspect.Signature:
     signature = inspect.signature(call)
     globalns = getattr(call, "__globals__", {})
-    typed_params = [
-        inspect.Parameter(
-            name=param.name,
-            kind=param.kind,
-            default=param.default,
-            annotation=get_typed_annotation(param, globalns),
-        )
-        for param in signature.parameters.values()
-    ]
+    if not inspect.isclass(call):
+        typed_params = [
+            inspect.Parameter(
+                name=param.name,
+                kind=param.kind,
+                default=param.default,
+                annotation=get_typed_annotation(param, globalns),
+            )
+            for param in signature.parameters.values()
+        ]
+    else:
+        from fastapi import Query
+        fields = {}
+        if call.__dict__.get("__fields__"):
+            for param in call.__fields__:
+                fields[param] = dict((call.__fields__[param].field_info.__repr_args__()))
+        typed_params = [
+            inspect.Parameter(
+                name=param.name,
+                kind=param.kind,
+                default=Query(fields[param.name].get("default"), description=fields[param.name].get("description")),
```

Comment:
What if the `fields` is empty? I recommend the following implementation

```py
# rest of code
default=Query(fields.get(param.name, {}).get("default"), description=fields.get(param.name, {}).get("description")
# rest of code
```

Reply:
thx for your review